### PR TITLE
"Default branch: latest" does not exist Fix.

### LINKED ISSUE #5390: "Default branch: latest" does not exist
* Read the Docs project: https://readthedocs.org/projects/test-builds/

Going to `Advanced settings` the `Default branch` selector shows `latest` as an option. I think this is invalid since the repo has no `latest` branch: https://github.com/rtfd/test-builds/branches/all?utf8=%E2%9C%93&query=latest

In this particular project seems that this is not a real problem, since the build that was triggered after saving it to this branch [picked the `origin/master` anyway](https://readthedocs.org/projects/test-builds/builds/8694899/) but I saw other projects failing for this (in the corporate site).

Anyway, `latest` should not be an option since it's not a valid branch on the repository.

### PR DESCRIPTION
**Pull request by saadmk11**
closes #5390

### CODE REVIEW

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def __init__(self, *args, **kwargs):
         # the special cases of LATEST and STABLE.
         all_versions_choices = [
             (v.commit_name, v.verbose_name)
```
Comment (stsewd) [inline readthedocs/projects/forms.py]: We can use `verbose_name` for the id too now (commit_name was used only for the special cases of latest and stable). Also, we should add a test when the user has a version named `latest` or `stable`. Using `commit_name` will make the test fail.

Something like https://github.com/rtfd/readthedocs.org/pull/5523/commits/82944cad1d1d7e6a300aa5cc55666196791d5950

Comment (saadmk11) [reply readthedocs/projects/forms.py]: Okay I'll Update the PR accordingly. :+1:

Comment (saadmk11) [reply readthedocs/projects/forms.py]: @stsewd What did you actually mean by this can you please clarify?

> Using commit_name will make the test fail.

Comment (stsewd) [reply readthedocs/projects/forms.py]: If you write a test with a tag named `latest` or `stable` the current implementation will fail.

Comment (stsewd) [reply readthedocs/projects/forms.py]: I mean, a version created by the user `machine=False`

Comment (saadmk11) [reply readthedocs/projects/forms.py]: @stsewd Updated the changes. but I'm Not sure I correctly understood what you meant about the tests. please have a look and let me know.

File: readthedocs/rtd_tests/tests/test_project_forms.py [hunk 2]
```diff
@@ -230,6 +230,15 @@ def setUp(self):
...
         )
+        get(
+            Version,
+            project=self.project,
+            slug='stable',
+            active=True,
+            privacy_level=PUBLIC,
+            identifier='ab96cbff71a8f40a4240aaf9d12e6c10',
+            verbose_name='stable',
+        )
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_project_forms.py]: Creating a version named `stable` is a specific test, we shouldn't merge it with the rest.

Comment (saadmk11) [reply readthedocs/rtd_tests/tests/test_project_forms.py]: @stsewd Okay :+1: . Updated the PR

File: readthedocs/rtd_tests/tests/test_project_forms.py [hunk 3]
```diff
@@ -261,20 +261,45 @@ def test_list_only_active_versions_on_default_version(self):
...
-    def test_list_all_versions_on_default_branch(self):
+    def test_list_only_non_auto_generated_versions_on_default_branch(self):
+        user_created_stable_version = get(
+            Version,
+            project=self.project,
+            slug='stable',
+            active=True,
+            privacy_level=PUBLIC,
+            identifier='ab96cbff71a8f40a4240aaf9d12e6c10',
+            verbose_name='stable',
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_project_forms.py]: Can we move the test for `stable` after the _normal_ tests? Also, we need a test for a branch named `latest`. Maybe we can just put this in another test case.

Comment (saadmk11) [reply readthedocs/rtd_tests/tests/test_project_forms.py]: @stsewd as these tests are for ProjectAdvancedForm I think we should put it here. And as you said we can just add a new test method inside this test case for `stable` and `latest`. And should I put only the assertions for `commit_name` inside the new test method?

Comment (saadmk11) [reply readthedocs/rtd_tests/tests/test_project_forms.py]: @stsewd Updated the Tests. Let me know if this will work for you. :)

**Comment by saadmk11** [comment]
@stsewd Thanks for helping me. It was a Great Experience :)

**Pull request merged**